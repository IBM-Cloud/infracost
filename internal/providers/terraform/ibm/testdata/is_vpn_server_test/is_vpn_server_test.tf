
terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_is_vpc" "vpc" {
  name = "vpc"
}

resource "ibm_is_subnet" "subnet_1" {
  name            = "subnet-1"
  vpc             = ibm_is_vpc.vpc.id
  zone            = "us-south-1"
  ipv4_cidr_block = "10.240.0.0/24"
}

resource "ibm_is_subnet" "subnet_2" {
  name            = "subnet-2"
  vpc             = ibm_is_vpc.vpc.id
  zone            = "us-south-2"
  ipv4_cidr_block = "10.240.0.0/24"
}

resource "ibm_is_vpn_server" "vpn_server" {
  name            = "vpn-server"
  certificate_crn = ""
  client_ip_pool  = "10.0.0.0/20"
  subnets         = [ibm_is_subnet.subnet_1.id]
  client_authentication {
    method            = "username"
    identity_provider = "iam"
  }
}

resource "ibm_is_vpn_server" "vpn_server_without_usage" {
  name            = "vpn-server-no-usage"
  certificate_crn = ""
  client_ip_pool  = "10.0.0.0/20"
  subnets         = [ibm_is_subnet.subnet_2.id]
  client_authentication {
    method            = "username"
    identity_provider = "iam"
  }
}
