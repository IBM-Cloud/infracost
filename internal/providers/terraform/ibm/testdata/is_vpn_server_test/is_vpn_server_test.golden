
 Name                                            Monthly Qty  Unit                Monthly Cost 
                                                                                               
 ibm_is_vpc.vpc                                                                                
 ├─ VPC instance                                           1  Instance                   $0.00 
 ├─ VPC egress free allowance (first 5GB)    Monthly cost depends on usage: $0.00 per GB       
 └─ VPC egress us-south (first 9995 GB)      Monthly cost depends on usage: $0.090915 per GB   
 └─ VPC egress us-south (next 40000 GB)      Monthly cost depends on usage: $0.086735 per GB   
 └─ VPC egress us-south (next 100000 GB)     Monthly cost depends on usage: $0.07315 per GB    
 └─ VPC egress us-south (over 149995 GB)     Monthly cost depends on usage: $0.05225 per GB    
                                                                                               
 ibm_is_vpn_server.vpn_server                                                                  
 ├─ VPN connection hours us-south                        730  Hours                      $7.63 
 └─ VPN instance hours us-south                          730  Hours                     $91.54 
                                                                                               
 ibm_is_vpn_server.vpn_server_without_usage                                                    
 ├─ VPN connection hours us-south            Monthly cost depends on usage: $0.01045 per Hours 
 └─ VPN instance hours us-south              Monthly cost depends on usage: $0.13 per Hours    
                                                                                               
 OVERALL TOTAL                                                                          $99.17 
──────────────────────────────────
5 cloud resources were detected:
∙ 3 were estimated, 1 of which usage-based costs, see https://infracost.io/usage-file
∙ 2 were free:
  ∙ 2 x ibm_is_subnet